Clarify Actions on nonzero Reserved Bits (#2280)
quicwg/base-drafts: draft-ietf-quic-transport.md

: The next two bits (those with a mask of 0x0c) of byte 0 are reserved.  These
  bits are protected using header protection (see Section 5.4 of {{QUIC-TLS}}).
  The value included prior to protection MUST be set to 0.  An endpoint MUST treat
  receipt of a packet that has a non-zero value for these bits after removing

>   packet (not just header) protection as a connection error of type

Reviewer: I don't think this is very clear to read. Do we really need to say anything, given that the timing side channel is described in detail in the TLS document?

Author: Oof. I didn't recall that section of the TLS draft. Thanks for pointing that out.

There are a couple of dangers here:
1) The implementer will not check "security considerations" in the tls draft when implementing first byte validation in the transport draft.
2) Naively, the implementation will choose to discard the packet and abort the connection after removing header protection, which opens up a DoS vector, even if goes through the motions of decrypting the packet anyway.

This is coming down to an editorial matter, and I agree my sentence is somewhat clunky. I'm open to other wording, but I don't think the current draft is satisfactory.

Reviewer 2: Let's start with an issue then, shall we?

Author: Aren't we down to wordsmithing at this point?

Reviewer 3: Isn't "packet protection" enough? I don't think you need to say "not just header"

Reviewer 2: It's both.  My suggestion would be:

```suggestion
  packet and header protection as a connection error of type
```

Reviewer 4: @martinthomson 's suggestion is good, but I'm confused why you need to remove packet protection to know the bits are not set to 0.  Isn't removing header protection sufficient?

  bits are protected using header protection (see Section 5.4 of
  {{QUIC-TLS}}).  The value included prior to protection MUST be set to 0.  An
  endpoint MUST treat receipt of a packet that has a non-zero value for these

>   bits after removing packet (not just header) protection as a connection error

Reviewer 2: ```suggestion
  bits after removing packet and header protection as a connection error
```
